app/models/feeds/megaphone_feed.rb

```diff
@@ -0,0 +1,61 @@
+class Feeds::MegaphoneFeed < Feed
+  DEFAULT_TITLE = "Megaphone Integration"
+
+  has_one :megaphone_config, class_name: "::Megaphone::Config", dependent: :destroy, autosave: true, validate: true, inverse_of: :feed
+
+  validate :must_have_token
+
+  after_initialize :set_defaults
+
+  alias_method :config, :megaphone_config
+
+  accepts_nested_attributes_for :megaphone_config, allow_destroy: true, reject_if: :all_blank
+
+  def paranoia_destroy_attributes
+    {
+      deleted_at: current_time_from_proper_timezone,
+      slug: "#{slug}-#{Time.now.to_i}"
```

Comment:
What's this for?  Just to obfuscate any existing copies of the url in the wild?

I wrote an `UnlinkJob` thing to remove stuff from S3 when the slug changes.  Wonder if we should also do that on feed destroy?

Reply:
Oh, maybe this is unrelated.  Just doing the same thing as Apple - keep the `slug` available.  The feed base class should probably do that too - someday!

Reply:
yeah the slug has to be unique, so we have to move it out of the way when logically deleting it